crates/iota-indexer/src/models/objects.rs

```diff
@@ -446,6 +446,72 @@ impl StoredObject {
     }
 }
 
+/// Holds a granular collection of [`StoredObject`] fields.
+///
+/// This can be used to consume an iterator of [`StoredObject`]
+/// to construct the collection of the individual fields.
+///
+/// ## Examples
+///
+/// ```ignore
+/// use iota_indexer::models::objects::{StoredObject, StoredObjects};
+/// fn construct_data() -> Vec<StoredObject> {
+///     Default::default()
+/// }
+///
+/// let mut stored_objects = StoredObjects::default();
+/// stored_objects.extend(construct_data());
+/// ```
+///
+/// Or using [`unzip`][std::iter::Iterator::unzip]:
+///
+/// ```ignore
+/// use iota_indexer::models::objects::{StoredObject, StoredObjects};
+/// use iota_types::digests::TransactionDigest;
+///
+/// fn construct_data() -> Vec<(StoredObject, TransactionDigest)> {
+///     Default::default()
+/// }
+///
+/// let (objects, digests): (StoredObjects, Vec<_>) = construct_data().into_iter().unzip();
+/// ```
+#[derive(Debug, Clone, Default)]
+pub(crate) struct StoredObjects {
```

Comment:
Seems the only purpose of this type is to simplify `persist_changed_objects`

Could we place it in the same file as `persist_changed_objects`? To treat it more like helper struct rather than general purpose model, since it's usage is quite specific.

Reply:
It is a helper struct indeed, but it is also essential to storing mutliple objects using raw queries. In this sense it is used a "model" of multiple objects for the write path, and hence I believe that it fits the scope of this module.

It also relates closely to `StoreObject` as well, and maintaining them together seems rather intuitive.